src/v/cluster/inventory_service.cc

```diff
@@ -0,0 +1,305 @@
+
+    vlog(cst_log.info, "Attempting to create inventory configuration");
+    if (const auto res = co_await _ops.maybe_create_inventory_configuration(
+          _remote->ref(), rtc);
+        res.has_error()) {
+        vlog(
+          cst_log.warn,
+          "Inventory configuration creation failed, will retry later",
+          res.error());
+        // If we failed creating inventory, try again later (on next expiry
+        // of _inventory_report_check_interval). If another node succeeded in
+        // creating the inventory, this is not counted as a failure.
+        _try_creating_inv_config = true;
+    } else {
+        vlog(
+          cst_log.info,
+          "Inventory configuration creation result: {}",
+          res.value());
+    }
+
+    _report_check_timer.set_callback([this] {
+        ssx::spawn_with_gate(_gate, [this] {
+            return check_for_current_inventory().handle_exception(
+              [](auto eptr) {
+                  vlog(
+                    cst_log.warn,
+                    "failed to check for current inventory: {}",
+                    eptr);
+              });
+        });
+    });
+
+    _report_check_timer.arm_periodic(_inventory_report_check_interval);
+}
+
+ss::future<> inventory_service::check_for_current_inventory() {
+    auto h = _gate.hold();
+    if (_try_creating_inv_config) {
+        auto rtc = retry_chain_node{_as, 60s, 1s};
+        auto res = co_await _ops.maybe_create_inventory_configuration(
+          _remote->ref(), rtc);
+
+        if (res.has_value()) {
+            _try_creating_inv_config = false;
+        }
+
+        // We either created the inventory just now, or failed again. In either
+        // case the report will not exist for several hours.
+        co_return;
+    }
+
+    // TODO use config values from cloud_storage
+    auto rtc = retry_chain_node{_as, 60s, 1s};
+    auto res = co_await _ops.fetch_latest_report_metadata(_remote->ref(), rtc);
+    if (res.has_error()) {
+        vlog(cst_log.info, "failed to fetch report metadata: ", res.error());
+        co_return;
+    }
+
+    const auto& [metadata_path, report_paths, datetime] = res.value();
+
+    // The ISO-8601 datetime format can be lexically compared
+    vlog(
+      cst_log.info,
+      "last fetched: {}, current report: {}",
+      _last_fetched,
+      datetime);
+    if (!_last_fetched.has_value() || datetime > _last_fetched.value()) {
+        begin_processing();
+        co_await clean_up_hash_directory();
+        if (co_await download_and_process_reports(report_paths)) {
+            vlog(
+              cst_log.info,
+              "finished processing inventory for report {}",
+              datetime);
+            mark_processing_complete();
+            _last_fetched = datetime;
+        }
+    }
+}
+
+ss::future<bool>
+inventory_service::download_and_process_reports(csi::report_paths paths) {
+    auto h = _gate.hold();
+    const auto ntps = co_await _leaders->ntps(_as);
```

Comment:
what will happen in case of leadership transfer?
some new ntps may become leaders after this check

Reply:
Since this data is generated on a best effort basis and the scrubber is supposed to account for leadership transfers:

* if we do not generate hashes for an NTP for which we became leader recently, the ntp hash loader will find no files for it. the scrubber is expected to skip segment checks and wait until next run.
* if we geenerate extra hashes IE an NTP leadership moved away after we created files for it, the files will be unused and removed on the next check iteration